manifests/virtualization/vmware.pp

```diff
@@ -0,0 +1,42 @@
+# vmware.pp
+# Install VMware Workstation Player for OS X, Ubuntu, or Windows
+# https://www.vmware.com/products/workstation-player.html
+#
+#
+
+class software::virtualization::vmware (
+  $ensure  = $software::params::software_ensure,
+  $version = $software::params::vmware_version,
+  $urlbase = $software::params::vmware_urlbase,
+  $package = $software::params::vmware_package,
+) inherits software::params {
+
+  validate_string($ensure)
+
+  case $::operatingsystem {
+    'Debian', 'Ubuntu': {
+      validate_string($version, $urlbase, $package)
```

Comment:
Can we switch to puppet 4 type validations?

Reply:
What would that look like? (Sorry!)

Reply:
Ah, type hinting, okay. That's the somewhat

```puppet
class software::virtualization::vmware (
  Enum['present', 'absent'] $ensure = $software::params::software_ensure,
  String $version = $software::params::vmware_version,
  String $url = $software::params::vmware_urlbase,
  String $package = $software::params::vmware_package,
) inherits software::params {
```
There may be [some types](https://www.example42.com/2015/09/09/puppet4-examples-data-types/) better suited for (semantic) version and URL than `String`, and `Optional[]` probably needs to be wrapped around some. What's your preference?

Hmmm ... (I'm trying to avoid the "I'm away from Puppet, currently" argument) :smirk: :zipper_mouth_face: :flushed:

Reply:
Stdlib has some types, but other than that I usually stick with the built in ones.
https://github.com/puppetlabs/puppetlabs-stdlib/tree/master/types